framework/contrib/hit/parse.cc

```diff
@@ -142,14 +142,15 @@ pathNorm(const std::string & path)
 std::string
 pathJoin(const std::vector<std::string> & paths)
 {
-  std::string fullpath;
+  static std::string fullpath;
```

Comment:
Is this the best solution you can think of? We generally like to avoid writable static variables, since they are not thread-safe and they also sometimes start misbehaving when we get to multiapps where we need multiple instantiations of objects that we normally only have one instance of. This particular usage doesn't look like it'll suffer from either of those issues but you still might consider a class member variable instead. I don't know if we have any other method level statics in MOOSE anywhere. Just trying to be consistent and clean.

Reply:
This particular mod was only a couple percentish - I'm also not a huge fan (not concurrency-safe among other things) and am okay with rolling it back if you think.  But this is just a plain function - not a member function - there is no class to put anything on.

Reply:
rolled back.